stl/inc/chrono

```diff
@@ -5999,6 +5999,20 @@ namespace chrono {
                 }
                 _Os << _Time.tm_mday;
                 return true;
+            case 'r':
+                if constexpr (_Is_specialization_v<_Ty, hh_mm_ss>) {
+                    // put_time uses _Strftime in order to bypass reference-counting that locale uses. This function
+                    // takes the locale information by pointer, but the pointer (from _Gettnames) returns a copy.
+                    // _Strftime delegates to other functions but eventually (for the C locale) has the %r specifier
+                    // rewritten. It checks for the locale by comparing pointers, which do not compare equal as we have
+                    // a copy of the pointer instead of the original. Therefore, we replace %r for the C locale
+                    // ourselves.

```

Comment:
If we're reporting this as a UCRT bug (which I think was our conclusion in the internal chat), should this be marked as `// TRANSITION, ...` with a bug number?  Theoretically, we won't have to special-case this forever!

Reply:
It's not exactly a UCRT bug. They can't really tell we mean the C locale when we pass it in through `_Strftime` because they are fundamentally different pointers. And we are the ones trying to circumvent the ref counting of locale. Maybe we can fix `time_put`? People might rely on the `%r` behavior though...